packages/xo-server/src/api/xostor.mjs

```diff
@@ -164,7 +164,13 @@ export const create = defer(async function (
 
   const task = await this.tasks.create({ name: `creation of XOSTOR: ${name}`, type: 'xo:xostor:create' })
   return task.run(async () => {
-    const hostIds = Object.keys(disksByHost)
+    const hostIds = Object.keys(disksByHost).filter(id => {
+      const nDisks = disksByHost[id].length
+      if (nDisks === 0) {
+        delete disksByHost[id]
+      }
+      return nDisks > 0
+    })
```

Comment:
Is it necessary to remove the entry from the collection?
```suggestion
    const hostIds = Object.keys(disksByHost).filter(hostId => disksByHost[hostId].length > 0)
```

Reply:
IMO, this isn't really necessary, but since the variable name is `disksByHost`, it's safer to remove entries that have an empty array.

Reply:
Sounds good. Then let's do something like this because it feels a bit weird to delete and filter at the same time:
```suggestion
    Object.entries(disksByHost).forEach(([hostId, disks]) => {
      if (disks.length === 0) {
        delete disksByHost[hostId]
      }
    })

    const hostIds = Object.keys(disksByHost)
```